### DISCUSSION TITLE
Feasibility of jupyterbook with matlab/octave

### DISCUSSION THREAD

**Discussion by joelmiller** [executablebooks/meta#1093] [Q&A]
I'm teaching a second year mathematics course with students who have been taught some matlab.  We will be writing a new set of notes, and I would like to do that through jupyterbook.  Is there a way to get matlab (or octave) to work with it?

I've seen suggestions that this is doable with R, but matlab/octave is what I need.

thanks,
Joel

**Comment by rgaiacs** [comment]
Hi @joelmiller,

Jupyter has a Octave kernel, https://github.com/jupyter-xeus/xeus-octave/ (you can read the announcement at https://blog.jupyter.org/a-jupyter-kernel-for-gnu-octave-b6d29e56341f). My understanding is that if you have the Octave kernel configured on the machine that is creating the book and set the Octave kernel in the Jupyter Notebooks, you should be able to write your project.

**Comment by joffreyp** [comment]
Not to dig this back up, but I recently had this issue, and wanted to be able to build a book featuring example Notebooks for Matlab and R in addition to Python. The only way through I've figured out is to use [Jupyter-Cache](https://jupyter-cache.readthedocs.io/en/latest/index.html) to directly cache pages after running them manually with appropriate kernels. Then I built the book using caches and defaulting to only building explicitly-referenced resources. This isn't ideal from an automation standpoint, but it will hopefully work for a slowly-changing demonstration book with examples in multiple languages.
